edge-hub/src/Microsoft.Azure.Devices.Routing.Core/Dispatcher.cs

```diff
@@ -48,29 +48,28 @@ public class Dispatcher : IDisposable
 
         ImmutableDictionary<string, IEndpointExecutor> Executors => this.executors;
 
-        public static async Task<Dispatcher> CreateAsync(string id, string iotHubName, ISet<Endpoint> endpoints, IEndpointExecutorFactory factory)
+        public static async Task<Dispatcher> CreateAsync(string id, string iotHubName, IDictionary<Endpoint, IList<uint>> endpointsWithPriorities, IEndpointExecutorFactory factory)
```

Comment:
talked offline, this is a list of priorities accumulated from multiple routes targeting the same endpoint, so that the executor knows how many underlying queues/checkpointers to maintain.

---
In reply to: [381807857](https://github.com/Azure/iotedge/pull/2525#discussion_r381807857) [](ancestors = 381807857)